Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h

```diff
@@ -272,21 +273,31 @@ class GstMappedFrame {
         return &m_frame.info;
     }
 
-    int width()
+    int width() const
     {
         return m_isValid ? GST_VIDEO_FRAME_WIDTH(&m_frame) : -1;
     }
 
-    int height()
+    int height() const
     {
         return m_isValid ? GST_VIDEO_FRAME_HEIGHT(&m_frame) : -1;
     }
 
-    int format()
+    int format() const
     {
         return m_isValid ? GST_VIDEO_FRAME_FORMAT(&m_frame) : GST_VIDEO_FORMAT_UNKNOWN;
     }
 
+    void* planeData(uint32_t planeIndex) const
+    {
+        return m_isValid ? GST_VIDEO_FRAME_PLANE_DATA(&m_frame, planeIndex) : nullptr;
+    }
+
+    int planeStride(uint32_t planeIndex) const
+    {
+        return m_isValid ? GST_VIDEO_FRAME_PLANE_STRIDE(&m_frame, planeIndex) : -1;
```

Comment:
I don't have a strong opinion about this but I would have made a different call here. I understand -1 is the "no stride option" for skia but in the case of the API we are defining here I would make this as `std::optional<int> planeStride...` and when calling this use `planeStride(x).value_or(-1)`

Reply:
No, this has nothing to do with Skia, this will never be used from Skia when it's invalid. I just followed what the other methods do for consistency. I think we shouldn't handle the invalid case, it's a programmer error to call any method on an invalid mapped frame, so I would just ASSERT(m_isValid) instead in all the methods.